### DISCUSSION TITLE
TCP/WS connection events handler

### DISCUSSION THREAD

**Discussion by kirankumar-ravuri** [baresip/re#761] [Q&A]
Hi All
I am trying libre in macOS. I am trying to understand whether there is any handler that gets called when TCP/WS connection is abruptly closed by the server. I don't seem to find any.  
Any pointers are appreciated.

**Comment by sreimers** [comment]
```c
int websock_connect(struct websock_conn **connp, struct websock *sock,
		    struct http_cli *cli, const char *uri, unsigned kaint,
		    websock_estab_h *estabh, websock_recv_h *recvh,
		    websock_close_h *closeh, void *arg,
		    const char *fmt, ...)
```
Normaly `websock_close_h *closeh` should be called.

**Comment by kirankumar-ravuri** [comment]
@sreimers I checked `websock_connect`, but this seems to be an internal api. 
I was actually asking from the application layer's perspective.  Application layer may want of retry/reconnect in case if there is any disconnection.
Is there any such api similar to sip_transp_add/sip_transp_add_websock, where application can get some callback on connection events?

**Comment by sreimers** [comment]
Can you provide a full code example, its hard to guess which APIs you use.

**Comment by kirankumar-ravuri** [comment]
@sreimers At the moment, I am using sip_transp_add_websock, sipreg_alloc, sipreg_unregister, sipsess_connect/info/abort apis.
